### PR TITLE
mac feature cleanup and cdrom 

### PR DESCRIPTION
**Pull request by gambol99**
- changed the mac address code to be more inline with the rest of the method 
- note: it would be nice if the schema validator in the vcloud-launch supported a regex check (though not a buggie)
- added the ability to attach cdroms on the fly to the vm
- @todo: a the moment the media is query uses just the vdc and media name, it's might an idea to add the catalog as well into the query, which is easily grabbed from the parent vapp

### CODE REVIEW

**Comment by ajlanghorn** [comment]
@gambol99 As mentioned previously, it would be good if you could clean this branch up so that the commits tell a story, as mentioned in @annashipman's [blog post]: http://www.annashipman.co.uk/jfdi/good-pull-requests.html. You'll also need to look at adding tests since our minimum threshold for feature completeness in this repository is 84%, and this pull request would mean we dip below that.

**Comment by surminus** [comment]
Hi @gambol99. We have cherry-picked the mac address logic and applied to push this into master (#173).

File: lib/vcloud/core/fog/service_interface.rb [hunk 1]
```diff
@@ -66,6 +66,16 @@ def get_vapp(id)
...
           end
 
+          def put_insert_cdrom(vm_id, media_id)
+            Vcloud::Core.logger.debug("inserting #{media_id} cdrom into VM #{vm_id}")
+            @vcloud.post_insert_cd_rom(vm_id, media_id)
+          end
+
+          def put_detach_cdrom(vm_id, media_id) 
+            Vcloud::Core.logger.debug("detaching #{media_id} cdrom into VM #{vm_id}")
+            @vcloud.post_insert_cd_rom(vm_id, media_id)
```
Comment (surminus) [inline lib/vcloud/core/fog/service_interface.rb]: @gambol99 Should this be using the post_eject_cd_rom method instead?

**Comment by gambol99** [comment]
Hi @surminus .. sorry, I fixed the typo for the detach cdrom method

**Comment by ajlanghorn** [comment]
@gambol99 Great, thanks. Please can you review the commits per [my earlier comment](https://github.com/gds-operations/vcloud-core/pull/169#issuecomment-111084342) before we look to merge this?

File: lib/vcloud/core/vm.rb [hunk 2]
```diff
@@ -200,6 +223,26 @@ def virtual_hardware_section
...
       end
 
+      def get_media_id_by_name(media_name)
+        media_id = get_media_href_by_name(media_name)
+        media_id.split("/").last
+      end
+
+      def get_media_href_by_name(media_name)
+        q = Vcloud::Core::QueryRunner.new 
+        vdc_results = q.run('vApp', :filter => "name==#{vapp_name}")
```
Comment (alexmuller) [inline lib/vcloud/core/vm.rb]: I'm not sure this query will work because `vapp_name` is undefined in this method.

**Comment by mattbostock** [comment]
Are you able to provide an update on this @gambol99? We'd love to get it merged once the remaining comments have been addressed.

I'm going to close this PR for now as it hasn't had an update in some time, however please feel free to raise a new PR when you're ready.
